source/cppfront.cpp

```diff
@@ -191,7 +191,7 @@ class positional_printer
         //  and where the last one was which determines our current colno
         auto last_newline = std::string::npos;  // the last newline we found in the string
         auto newline_pos  = 0;                  // the current newline we found in the string
-        while ((newline_pos = s.find('\n', newline_pos)) != std::string::npos)
+        while ((newline_pos = s.find('\n', newline_pos)) != static_cast<ssize_t>(std::string::npos))
```

Comment:
I think it'd be better to change the `0` above to `0z`.

Reply:
`find` returns `std::string::size_type`, so that should be the type of `newline_pos`.

Reply:
> Probably the easiest way to do that is to initially assign it `std::string::npos` instead of `0`.

That seems to be the optimal solution here. Either that or explicitly specifying the type to avoid the needless cast. The `static_cast` to `ssize_t` isn't even fully correct either, since the deduced type of `newline_pos` would be `int`, while `std::string::size_type` is `std::size_t`.

Reply:
>That seems to be the optimal solution here. Either that or explicitly specifying the type 

I chose that because I believe that Herb prefers `auto var = type{val};` over `type var = val;` even in C++1.

Reply:
> I chose that because I believe that Herb prefers `auto var = type{val};` over `type var = val;` even in C++1.

I personally do prefer that too, but here it should then be `auto newline_pos = std::string::size_type{0}`. This way 
`newline_pos` will not be deduced to `int`, and the `static_cast` can be removed too.

Reply:
@switch-blade-stuff That's equivalent to `auto newline_pos = std::string::npos;` and yes that is a replacement for the static_cast later.

Reply:
@gregmarr if I'm correct, `npos` is `-1` instead, but it will have the same effect, yes, since it is re-assigned immediately either way.

Reply:
Yes, that is all correct.

Reply:
I would suggest to remove line 193 altogether and limit the scope of the variable to only where it is used:
```suggestion
        while ((auto newline_pos = s.find('\n', newline_pos)) != std::string::npos)
```

Reply:
I don't think that's in the grammar.

Reply:
Oh, sure the `newline_pos` is actually used in the `find(...)` 🤦.

Reply:
Oh, true, the extra parens... and without the parens it would not work even it wasn't for the issue with the `find`. Lost cause anyway. Didn't save before compiling :D.
I agree with the last proposition of @switch-blade-stuff: `auto newline_pos = std::string::size_type{0}`. The initial value needs to be 0 as it is used for the first call to `find`.

Reply:
>The initial value needs to be 0 as it is used for the first call to find.

Good catch, I missed that.